### ISSUE TITLE
Thread-local LocalScheduler$Activation objects are never garbage-collected

### ISSUE THREAD

**Issue by y0un5** [twitter/util#118]
LocalScheduler has thread-local Activation objects.

```scala
class LocalScheduler(lifo: Boolean) extends Scheduler {
  ...
  @volatile private[this] var activations = Set[Activation]()
  private[this] val local = new ThreadLocal[Activation] {
    override def initialValue = null
  }
```

And when a thread-local Activation is created, it is added to activations var. 

```scala
private class Activation extends Scheduler with Iterator[Runnable] {
  ...
  private[this] def get(): Activation = {
    ...
    local.set(new Activation)
    synchronized { activations += local.get() }
    local.get()
  }
```

But there is no code to remove Activation objects from activations. So these are never garbage-collected, because they are referenced by activations even after their owner threads are terminated.

This is not a problem if using only fixed-size threads, but if using a dynamic thread pool, it can lead to a memory leak.

**Comment by kevinoliver** [comment]
Thanks for the report. 

We are reviewing a commit that would fix the leak.

**Comment by kevinoliver** [comment]
Sorry for the late update. A fix for this has been applied locally and should show up on the develop branch in the near future.

**Comment by vkostyukov** [comment]
This has been fixed by 32f16db.
